psydac/api/discretization.py

```diff
@@ -271,7 +271,11 @@ def discretize_space(V, domain_h, *, degree=None, multiplicity=None, knots=None,
     connectivity = construct_connectivity(domain)
     if isinstance(domain_h, Geometry) and all(domain_h.mappings.values()):
         # from a discrete geoemtry
-        mappings  = [domain_h.mappings[inter.logical_domain.name] for inter in interiors]
+        if interiors[0].name in domain_h.mappings:
+            mappings  = [domain_h.mappings[inter.name] for inter in interiors]
+        else:
+            mappings  = [domain_h.mappings[inter.logical_domain.name] for inter in interiors]
+
```

Comment:
Can you explain this change?

Reply:
In some cases, the mappings dictionary has the logical domain interiors as a keyword and in some other cases the physical domain interiors.

Reply:
This suggests that we have a fundamental inconsistency (i.e. a bug) somewhere in our code base. If you can describe when the two situations happen, please create an issue!

Reply:
done https://github.com/pyccel/psydac/issues/264

Reply:
Thanks!